core/src/main/java/org/opensearch/sql/expression/aggregation/AvgAggregator.java

```diff
@@ -23,19 +30,41 @@
  */
 public class AvgAggregator extends Aggregator<AvgAggregator.AvgState> {
 
+  /**
+   * To process by different ways different data types, we need to store the type.
+   * Input data has the same type as the result.
+   */
+  private final ExprCoreType dataType;
+
   public AvgAggregator(List<Expression> arguments, ExprCoreType returnType) {
     super(BuiltinFunctionName.AVG.getName(), arguments, returnType);
+    dataType = returnType;
   }
 
   @Override
   public AvgState create() {
-    return new AvgState();
+    return new AvgState(dataType);
   }
 
   @Override
   protected AvgState iterate(ExprValue value, AvgState state) {
     state.count++;
-    state.total += ExprValueUtils.getDoubleValue(value);
+    switch (dataType) {
+      case DATE:
+      case DATETIME:
+      case TIMESTAMP:
+        state.total += value.timestampValue().toEpochMilli();
+        break;
+      case TIME:
+        state.total += MILLIS.between(LocalTime.MIN, value.timeValue());
+        break;
+      case DOUBLE:
```

Comment:
we cannot compute AVG for SHORT, INTEGER, FLOAT, etc...?

Reply:
These types are casted to DOUBLE automatically